src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -2426,7 +2443,27 @@ private void DisposeIndexWriterOnError(Lazy<IndexWriteOperationBase> writeOperat
                     writeOperation.Value.Dispose();
             }
         }
+        
+        private Dictionary<string, bool> _collectionTracker;
+        
+        public IIndexedItemEnumerator GetMapEnumeratorAndBoxWhenNeeded(IEnumerable<IndexItem> items, string collection, TransactionOperationContext indexContext,
+            IndexingStatsScope stats, IndexType type)
+        {
+            var enumerator = GetMapEnumerator(items, collection, indexContext, stats, type);
+            
+            if (IsTestRun)
+            {
+                _collectionTracker ??= new();
+                if (_collectionTracker.ContainsKey(collection))
+                    return new EmptyItemEnumerator();
+                
+                _collectionTracker[collection] = true;
+                return new BoxedItemEnumerator(enumerator, Collections.Count);
```

Comment:
Can this logic be inside TestIndexRun? You could have a similar method to `CreateIndexWriteOperationWrapper()`

Reply:
I moved the logic to TestRun.CreateEnumerator